x-pack/plugins/security_solution/public/detections/components/rules/rule_preview/index.tsx

```diff
@@ -63,21 +65,33 @@ const RulePreviewComponent: React.FC<RulePreviewProps> = ({
   anomalyThreshold,
 }) => {
   const { spaces } = useKibana().services;
+  const { loading: isMlLoading, jobs } = useSecurityJobs(false);
+
   const [spaceId, setSpaceId] = useState('');
   useEffect(() => {
     if (spaces) {
       spaces.getActiveSpace().then((space) => setSpaceId(space.id));
     }
   }, [spaces]);
 
+  const areAllMlJobsRunning = useMemo(() => {
+    if (ruleType !== 'machine_learning') {
+      return true; // Don't do the expensive logic if we don't need it
```

Comment:
maybe we could consider renaming the variable to `isRelevantMLJobsRunning` so returning `true` for non-ML rules makes more sense. Or maybe we can rename it to `isPreviewDisabled` and add `isDisabled` bool in here as well, so all of the disable logic gets encapsulated in this block